### ISSUE TITLE
Fatal Error when uri = post or post/

### ISSUE THREAD

**Issue by guangrei** [guzzle/guzzle#1951]
| Q            | A
| ------------ | ---
| Bug?         | yes
| New Feature? | no|yes
| Version      | ^6.3

Hi, i used guzzle for sending post to https://url/api/post/ 
but it doesn't seem to work

here for mycode
``` php
$r = new \GuzzleHttp\Client(['base_uri'=>'https://url/api/']);
$r->request('POST','post',[// array parameter]);
```
result :
```
Fatal error: Magic request methods require a URI and optional options array
```

then i made change to  `post/` 
```
$r->request('POST','post/',[// array parameter]);
```

got same error result, once again with  `/post` 
```
$r->request('POST','/post',[// parameter]);
```

 **work but 404** 

and try another string e.g :  `test` etc all work except  `post` and  `post/` 

---
PHP VERSION : 7.2.0alpha1

**Comment by alexeyshockov** [comment]
Could you try to run your code on a stable version of PHP?

A working example would be also great to have (at the moment the bug is not reproducible, not so many chances to get help).

**Comment by guangrei** [comment]
solved in php 7.2 release, thanks :)
